spec/requests/payments_controller_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe "/payments/:id/authorize" do
+  let!(:user) { create(:user) }
+  let!(:order) { create(:order, user:) }
+  let!(:payment) { create(:payment, order:) }
+
+  describe "when user isn't logged in" do
+    it "redirects to the login page and set error flash msg" do
+      get authorize_payment_path(payment)
+      expect(response).to redirect_to(root_path(anchor: "/login", after_login: request.fullpath))
+      expect(flash[:error]).to eq I18n.t("spree.orders.edit.login_to_view_order")
```

Comment:
FYI We generally use the translated text in specs (using default translation `en.yml`). Although it creates slightly more work, it creates a more robust test (it's clearer what message should be communicating, and can help test things like interpolation). But this is fine as-is.

https://github.com/openfoodfoundation/openfoodnetwork/blob/57f148564ceeb229acdba43b8049af5be6a97770/config/locales/en.yml#L4566